### PR TITLE
Add a note about the open(2) and rename(2) on AUFS and OverlayFS

### LINKED ISSUE #25409: os.rename throws exception when renaming directories on lower overlayfs layers
**Output of `docker version`:**

```
Client:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:        Thu Jul 28 22:11:10 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:        Thu Jul 28 22:11:10 2016
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 2
 Running: 0
 Paused: 0
 Stopped: 2
Images: 29
Server Version: 1.12.0
Storage Driver: overlay2
 Backing Filesystem: extfs
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: host bridge null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: seccomp
Kernel Version: 4.4.7-1.ng
Operating System: Ubuntu 16.04 LTS
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 15.67 GiB
Name: rovner-qtesting-7
ID: M7UX:OR2K:WJQI:KGPM:XFIJ:UF72:AQ6J:OYUZ:UIBO:6RJZ:LVGN:TT4I
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
WARNING: No kernel memory limit support
Insecure Registries:
 127.0.0.0/8
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
```
root@my-host:/# uname -a
Linux my-host 4.4.7-1.ng #1 SMP Fri Apr 15 17:59:44 MSK 2016 x86_64 x86_64 x86_64 GNU/Linux
```

**Steps to reproduce the issue:**
- Build image with Dockerfile
```
FROM ubuntu:14.04
RUN mkdir /foo && touch /foo/bar
```

```
Sending build context to Docker daemon 2.048 kB
Step 1 : FROM ubuntu:14.04
 ---> 0ccb13bf1954
Step 2 : RUN mkdir /foo && touch /foo/bar
 ---> Running in dc97258d4d23
 ---> c6216cc93147
Removing intermediate container dc97258d4d23
Successfully built c6216cc93147
```

- Rename foo to foo2 with python `os.rename`
```
root@c18071cd5033:/# python3
Python 3.4.3 (default, Oct 14 2015, 20:28:29)
[GCC 4.8.4] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> os.rename('/foo', '/foo2')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
OSError: [Errno 18] Invalid cross-device link: '/foo' -> '/foo2'
```

**Describe the results you received:**
Rename failed with error `OSError: [Errno 18] Invalid cross-device link: '/foo' -> '/foo2'`

**Describe the results you expected:**
Rename finished successfully

### PR DESCRIPTION
**Pull request by AkihiroSuda**
Close #25409

information source:
- Overlay https://github.com/torvalds/linux/blob/v4.8-rc2/fs/overlayfs/copy_up.c#L318-L322
- AUFS https://github.com/sfjro/aufs4-standalone/blob/29760abed58378541ec6b9fedb77310b5a012ef4/fs/aufs/i_op_ren.c#L452-L454

### CODE REVIEW

File: docs/userguide/storagedriver/overlayfs-driver.md [hunk 1]
```diff
@@ -413,3 +414,17 @@ performance. This is because they bypass the storage driver and do not incur
...
+
+## OverlayFS compatibility
+To summarize the OverlayFS's aspect which is incompatible with other
+filesystems:
+
+- **RPMs and Yum**. OverlayFS only implements a subset of the POSIX standards. 
+This can result in certain OverlayFS operations breaking POSIX standards. One 
+such operation is the *copy-up* operation. Therefore, using `yum` inside of a 
+container on a Docker host using the `overlay`/`overlay2` storage drivers is 
+unlikely to work without implementing workarounds.
```
Comment (thaJeztah) [inline docs/userguide/storagedriver/overlayfs-driver.md]: I think recent versions of yum are patched to work around this issue (but I don't know which version)

**Comment by thaJeztah** [comment]
Thanks, one question/note, but LGTM otherwise

File: docs/userguide/storagedriver/overlayfs-driver.md (hunk 1 above)
Comment (dmcgowan) [inline docs/userguide/storagedriver/overlayfs-driver.md]: Was this issue specifically related to the inode changing on a file which was open for read? I know this is not specifically related to this PR but may be a good chance to soften to the language, "unlikely to work" is no longer accurate.

**Comment by dmcgowan** [comment]
same question, and LGTM

File: docs/userguide/storagedriver/overlayfs-driver.md (hunk 1 above)
Comment (thaJeztah) [inline docs/userguide/storagedriver/overlayfs-driver.md]: ping @vbatts could you help us out here?

Comment (vbatts) [inline docs/userguide/storagedriver/overlayfs-driver.md]: I understand there to be fixes for this behavior now in place. And was it that it did not work? I think it _worked_ but had error output. Regardless, didn't look great. :-)

Comment (thaJeztah) [inline docs/userguide/storagedriver/overlayfs-driver.md]: Probably it's safe to remove the warning about yum then (thanks @vbatts)

**Comment by AkihiroSuda** [comment]
OK, updated PR

**Comment by vdemeester** [comment]
LGTM :cat:

**Pull request merged**